internal/controllers/topology/cluster/structuredmerge/serversidepathhelper.go

```diff
@@ -44,17 +44,8 @@ type serverSidePatchHelper struct {
 func NewServerSidePatchHelper(original, modified client.Object, c client.Client, opts ...HelperOption) (PatchHelper, error) {
 	helperOptions := &HelperOptions{}
 	helperOptions = helperOptions.ApplyOptions(opts)
-	helperOptions.allowedPaths = []contract.Path{
-		// apiVersion, kind, name and namespace are required field for a server side apply intent.
-		{"apiVersion"},
-		{"kind"},
-		{"metadata", "name"},
-		{"metadata", "namespace"},
-		// the topology controller controls/has an opinion for labels, annotation, ownerReferences and spec only.
-		{"metadata", "labels"},
-		{"metadata", "annotations"},
-		{"metadata", "ownerReferences"},
-		{"spec"},
+	if len(helperOptions.allowedPaths) == 0 {
+		helperOptions.ApplyOptions([]HelperOption{AllowedSpecPaths{allowedPathSpec}})
```

Comment:
This is kind of hard to follow, given that it is meant to trigger also adding metadata allowed paths